extra/end-to-end-test.rb

```diff
@@ -0,0 +1,66 @@
+require 'rest-client'
+require 'json'
+require 'pp'
+
+# subnets and domains must exist on Infoblox
+URL = ENV['FOREMAN_PROXY_URL'] || "http://localhost:8000"
+DNS_DOMAIN = ENV['IBX_DNS_DOMAIN'] || "example.com"
+DNS_IPV4 = ENV['IBX_DNS_IPV4'] || "192.0.2.1"
+DNS_IPV6 = ENV['IBX_DNS_IPV6'] || "2001:db8::1"
+DNS_PTR4 = ENV['IBX_DNS_PTR4'] || "1.2.0.192.in-addr.arpa"
+DNS_PTR6 = ENV['IBX_DNS_PTR6'] || "1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.8.b.d.0.1.0.0.2.ip6.arpa"
+
+def do_get(path)
+  RestClient.get("#{URL}#{path}")
+end
+
+def do_post(path, body = nil, params = {})
+  RestClient.post("#{URL}#{path}", body, params.merge({content_type: :json}))
+end
+
+def do_delete(path, params = {})
+  RestClient.delete("#{URL}#{path}", params.merge({content_type: :json}))
+end
+
+features = JSON.parse(do_get("/v2/features"))
+
+if features["dns"]["state"] == "running"
+  fqdn = "end2end.#{DNS_DOMAIN}"
+  cname = "alias-end2end.#{DNS_DOMAIN}"
+  puts "Performing DNS tests with #{fqdn}"
+  begin
+    do_post("/dns?type=a&fqdn=#{fqdn}&value=#{DNS_IPV4}")
+    do_post("/dns?type=a&fqdn=#{fqdn}&value=#{DNS_IPV4}") # skip (already exists)
+    do_post("/dns?type=a&fqdn=#{fqdn}&value=192.0.2.1") rescue RestClient::Conflict # expected
```

Comment:
My point was that if this doesn't raise an exception, it's masking a problem. There must be a conflict or you're violating the protocol. The current code can not raise an exception and pass. Since this is actually what's being changed, I'd like to see specific verification of this (since I don't have an infoblox instance to test with).

Reply:
I am not following, this should (and does) return HTTP 409 conflict, which makes RestClient to throw this exception.

Reply:
I'm talking about the other case. If it doesn't raise a conflict, it's a problem. So roughly:
```suggestion
    begin
        do_post("/dns?type=a&fqdn=#{fqdn}&value=192.0.2.1")
        raise Exception # We expect a conflict but didn't get one.
    rescue RestClient::Conflict
        # expected
    end
```

Reply:
Sure I am back. Done.